x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/convert_inference_events_to_streaming_events.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OperatorFunction, map } from 'rxjs';
+import { v4 } from 'uuid';
+import {
+  ChatCompletionEvent as InferenceChatCompletionEvent,
+  ChatCompletionEventType as InferenceChatCompletionEventType,
+} from '@kbn/inference-common';
+import {
+  ChatCompletionChunkEvent,
+  TokenCountEvent,
+  ChatCompletionMessageEvent,
+  StreamingChatResponseEventType,
+} from '../../../../common';
+
+export function convertInferenceEventsToStreamingEvents<
+  T extends InferenceChatCompletionEvent
+>(): OperatorFunction<T, ChatCompletionChunkEvent | TokenCountEvent | ChatCompletionMessageEvent> {
+  return (events$) => {
+    return events$.pipe(
+      map((event) => {
+        if (event.type === InferenceChatCompletionEventType.ChatCompletionChunk) {
+          // Convert to ChatCompletionChunkEvent
+          return {
+            type: StreamingChatResponseEventType.ChatCompletionChunk,
```

Comment:
nit: Should `StreamingChatResponseEventType` be renamed to `ChatCompletionEventType` to align with the inference naming?

```suggestion
            type: ChatCompletionEventType.ChatCompletionChunk,
```

Reply:
I agree that renaming for alignment makes sense, but before doing so, we should consider splitting the event types. The `StreamingChatResponseEventType` includes a broader range of events beyond just `ChatCompletion`:

``` typescript
export enum StreamingChatResponseEventType {
  ChatCompletionChunk = 'chatCompletionChunk',
  ChatCompletionMessage = 'chatCompletionMessage',
  ConversationCreate = 'conversationCreate',
  ConversationUpdate = 'conversationUpdate',
  MessageAdd = 'messageAdd',
  ChatCompletionError = 'chatCompletionError',
  BufferFlush = 'bufferFlush',
  TokenCount = 'tokenCount',
}
```

Reply:
Yes, that makes sense. Do you think that should be done in this PR or as a follow-up? I don't mind either